src/engraving/libmscore/deadslapped.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef __DEADSLAPPED_H__
+#define __DEADSLAPPED_H__
+
+#include "durationelement.h"
+
+namespace mu::engraving {
+//---------------------------------------------------------
+//    @@ DeadSlapped
+///     This class implements a dead slapped element.
+//---------------------------------------------------------
+
+class DeadSlapped : public EngravingItem
```

Comment:
General comment, @cbjeukendrup please correct me if I'm wrong here. Since DeadSlapped is, essentially, a rest, we could consider deriving this class from Rest or ChordRest. At the moment, if I understand correctly, DeadSlapped is defined as just a graphical element, and you are drawing it in place of the rest when you need it. That seems a bit hacky to me. Deriving it from Rest would let this class inherit lots of stuff that may turn out useful when we build proper support for this element in Musescore. Doesn't need to be done now, anyway, we can also do it later.